internal/gps/vcs_repo_test.go

```diff
@@ -46,6 +52,53 @@ func TestErrs(t *testing.T) {
 	}
 }
 
+func TestNewCtxRepo(t *testing.T) {
```

Comment:
just fyi, this particular test has no happy path. It's just the failure+recovery part. So it won't be a split per se. more like adding a new test that is a happy path one. Adding one : )